lib/trento/application/integration/checks/adapter/runner.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Trento.Integration.Checks.Runner do
+  @moduledoc """
+  Trento runner integration adapter
+  """
+
+  @behaviour Trento.Integration.Checks.Gen
+
+  @impl true
+  def request_execution(_execution_id, _cluster_id, _hosts, _selected_checks) do
+    :ok
+  end
+
+  @impl true
+  def get_catalog do
+    runner_url = runner_url()
+
+    case HTTPoison.get("#{runner_url}/api/catalog") do
+      {:ok, %HTTPoison.Response{status_code: 200, body: catalog}} ->
+        Jason.decode(catalog)
+
+      {:ok, %HTTPoison.Response{status_code: 204}} ->
+        {:error, :not_ready}
+
+      {:error, %HTTPoison.Error{reason: :econnrefused}} ->
+        {:error, "Connection to the runner component on #{runner_url} was refused."}
+
+      {:error, %HTTPoison.Error{reason: :nxdomain}} ->
+        {:error,
+         "Connection url to the runner component host on #{runner_url} could not be resolved."}
+
+      {:error, %HTTPoison.Error{reason: reason}} ->
+        {:error, reason}
+
```

Comment:
Do we need to match all this cases? just matching with error reason is enough and also we want to return :error, atom tuples and not strings

Reply:
And where do we "create" the error messages? In the frontend?
I wouldn't like to show a message like `econnrefused`

Reply:
Does the user need to know if there was an nslookup error or a connection refused error? For the frontend i would just bubble up that something went wrong and log it for debugging purposes.
Also, in general, is not responsability of an adapter to think about the FE since there could be no FE at all.

Reply:
Well, I personally find it much more user friendly to give specific error messages than "Something went wrong" message, if we know at least which was the error. And this is case, I find it useful that the error contains the runner url, it makes things easier.
Anyway, I don't want to open a battle about this here.
I will remove the messages and we will see what we do with them in the future

Reply:
Keep in mind that when running in kubernetes the runner URL will be always the service name .e.g http://trento-runner-service